### PR TITLE
Adding OneBodyDensityMatricesInput and tests

### PR DESCRIPTION
**Pull request by PDoakORNL**
## Proposed changes

Initial Input class for OneBodyDensityMatrices (the port of DensityMatrices1B) to the unified driver setup.

Using InputSection for parsing and simple validation.  Input class has native type and default declaration/definition.

## What type(s) of changes does this code introduce?
- New feature
- Documentation changes

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
Leconte

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes This PR is up to date with current the current state of 'develop'
- Yes Code added or changed in the PR has been clang-formatted
- Yes This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by PDoakORNL** [comment]
No idea why codecov patch indicates failure.  I increased coverage with this PR.

**Review by jtkrogel — APPROVED** [review]
The code here LGTM.  I'm not in love with the extra boilerplate in the input parsing, but it's good to have multiple proposals for this.

File: src/Estimators/InputSection.cpp [hunk 1]
```diff
@@ -158,7 +155,8 @@ void InputSection::check_valid()
             << " has not been assigned\n";
       throw UniformCommunicateError(error.str());
     }
-}
+  this->checkParticularValidity();
```
Comment (ye-luo) [inline src/Estimators/InputSection.cpp:158]: Why is `this->` needed? InputSection is not a template class.

File: src/Estimators/InputSection.h [hunk 2]
```diff
@@ -61,7 +61,7 @@ class InputSection
   // Enable read-only access to variable values.
   //   Needs updating to allow copy-less return.
   template<typename T>
-  T get(const std::string& name) const
+  const T get(const std::string& name) const
```
Comment (ye-luo) [inline src/Estimators/InputSection.h:64]: const has no effect when return by value.

File: src/Estimators/OneBodyDensityMatricesInput.cpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+// Some code refactored from: DensityMatrices1b.cpp
+//////////////////////////////////////////////////////////////////////////////////////
+
+#include "string_utils.h"
+#include "OneBodyDensityMatricesInput.h"
+
+namespace qmcplusplus
+{
+
+OneBodyDensityMatricesInput::OneBodyDensityMatricesInput(){};
```
Comment (ye-luo) [inline src/Estimators/OneBodyDensityMatricesInput.cpp:18]: `= default`

Comment (PDoakORNL) [reply src/Estimators/OneBodyDensityMatricesInput.cpp:18]: This as well

File: src/Estimators/InputSection.h (hunk 2 above)
Comment (PDoakORNL) [reply src/Estimators/InputSection.h:64]: will fix in later PR

File: src/Estimators/InputSection.cpp (hunk 1 above)
Comment (PDoakORNL) [reply src/Estimators/InputSection.cpp:158]: and this

**Comment by PDoakORNL** [comment]
Start testing in-house

**Review by ye-luo — APPROVED** [review]

**Pull request merged**